_posts/2021-01-13-hapi-fhir.md

```diff
@@ -391,64 +411,74 @@ The AuthorizationInterceptor is used by subclassing it and then registering your
-            // This user has access to everything
-            userIsAdmin = true;
-        } else {
-            // Throw an HTTP 401
-            throw  new  AuthenticationException("Missing or invalid Authorization header value");
-        }
-        // If the user is a specific patient, we create the following rule chain:
-        // Allow the user to read anything in their own patient compartment
-        // Allow the user to write anything in their own patient compartment
-        // If a client request doesn't pass either of the above, deny it
-        if (userIdPatientId != null) {
+        IdType userIdPatientId = null;
+        String authHeader = theRequestDetails.getHeader("Authorization");
+        if (isNormalUser(authHeader)) {
+            // This user has only access to the Patient resource with id 1.
+            // If the user is a specific patient, we create the following rule chain:
+            // Allow the user to read anything in their own patient compartment
+            // Allow the user to write anything in their own patient compartment
+            // If a client request doesn't pass either of the above, deny it
+            userIdPatientId = new IdType("Patient", 1L);
             return new RuleBuilder()
                     .allow().read().allResources().inCompartment("Patient", userIdPatientId).andThen()
                     .allow().write().allResources().inCompartment("Patient", userIdPatientId).andThen()
                     .denyAll()
                     .build();
-        }
-        // If the user is an admin, allow everything
-        if (userIsAdmin) {
+        } else if (isAdmin(authHeader)) {
+            // If the user is an admin, allow everything
             return new RuleBuilder()
                     .allowAll()
                     .build();
+        } else {
+            // Throw an HTTP 401
+            throw new AuthenticationException("Missing or invalid Authorization header value");
         }
-        // By default, deny everything. This should never get hit, but it's 
-        // good to be defensive
-        return new RuleBuilder()
-                .denyAll()
-                ´´.build();
     }
+
+    private boolean isNormalUser(String authHeader){
+        return "Bearer dfw98h38r".equals(authHeader);
+
+    }
+    private boolean isAdmin(String authHeader){
+        return "Bearer 39ff939jgg".equals(authHeader);
+    }
+
 }
 ```
 
-  
+After the server was set up and ready to use, I set up a small angular front-end application where it was possible to retrieve all the Patients and add a Patient.
+Finally observations were added which could be linked to a specific patient.
+In order not to define all the interfaces myself I used the following module [@ahryman40k/ts-fhir-types](https://www.npmjs.com/package/@ahryman40k/ts-fhir-types){:target="_blank",rel="noopener noreferrer"}.
 
 ## Conclusion
 The FHIR specification has been providing enhancements at various levels for several years, 
```

Comment:
suggest rephrase to improve readbility:
For several years, the FHIR specification has been providing enhancements at various levels 
from application development to inter-application integration.